src/js_native_api_v8.h

```diff
@@ -25,6 +25,9 @@ struct napi_env__ {
 
   virtual bool can_call_into_js() const { return true; }
 
+  virtual void track_finalizer(void (*finalizer)(void*), void* data) {}
+  virtual void untrack_finalizer(void (*finalizer)(void*), void* data) {}
```

Comment:
style nit that can totally be ignored: These methods would more typically be called `TrackFinalizer` and `UntrackFinalizer` since they aren’t simple getters/setters. 

Going even further, I think *every* implementor of N-API should support `napi_add_env_cleanup_hook()` and `napi_remove_env_cleanup_hook()`; they could totally be no-ops for some non-Node.js variants that do follow a one-env-per-process model like Node.js formerly did, but the *concept* of “end of execution of the current runtime instance” is not specific to Node.js and addons should handle it properly, regardless of the choice of the runtime.

If those methods are moved to `js_native_api.h`, we could use them directly instead, which might be nice.

Reply:
I guess the separation bethttps://github.com/nodejs/nodeween `node_api.h` and `js_native_api.h` is not 100% clear. The initial thrust was to separate APIs related to JS values from ones related to the environment. Runtime-specific things like AsyncWorker and TSFN have thus also ended up in `node_api.h`.

Reply:
> The initial thrust was to separate APIs related to JS values from ones related to the environment.

I get that, but I would argue that because Agents are defined in ECMA-262, management of an Agent’s lifetime is also implicitly part of JS ;) Also, like I said, if runtimes don’t support any kind of cleanup, that’s fine and they can just make the functions have empty bodies.

Reply:
@addaleax I'll talk it over with @nodejs/n-api and make that into a separate PR.